Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1

```diff
@@ -31,6 +31,35 @@ function Format-Splat {
     "$parameters" # String join the array.
 }
 
+function Generate-TlsError {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)] $exception 
+    )
+
+    if ($exception -eq $null)
+    {
+        return
+    }
+
+    $isWebException = $true
+    $innerException = $exception
+    while ($innerException.GetType() -ne [System.Net.WebException])
+    {
+        $innerException = $innerException.InnerException
+        if ($innerException -eq $null)
+        {
+            $isWebException = $false
+            break
+        }
+    }
+
+    if (($isWebException -eq $true) -and ($innerException.InnerException -ne $null) -and ($innerException.InnerException.GetType() -eq [System.IO.IOException]))
```

Comment:
is IOException always going to be due to TLS error?

Reply:
Probably not. But that is our only hint that it might be that. We are checking for WebException whose inner exception is IOException. If that is the case, it is probably TLS.